fix: enable tenant-aware search by default for OpenSearch

Christian-Broomfield opened the pull request:
## Description

At the moment, processes from deleted tenants appear in search results on the /v1/process-instances/search endpoint when running with Opensearch. This is due to the OpenSearch DAOs overwriting the `SearchRequest.query` query without adding `withTenantCheck`. This PR fixes the mistake and structures the aggregation of the filter query so that such errors are less likely to happen in the future.

## Checklist

- [x] Enable backports for 8.7 and 8.8

## Related issues

closes #38297

Christian-Broomfield committed:
fix: /v1/process-instance/search not tenant-aware with filter queries

mustafadagher reviewed (changes requested):
Thanks a lot for the effort you made here @Christian-Broomfield 🙏 

❌ While I know you tried to follow the direction that was defined in the original ticket, I believe that the main problem is applying a `wildcard tenant check` by default [here](https://github.com/camunda/camunda/blob/ff8dbe135d1523283ba1324ed42c98824150432d/operate/webapp/src/main/java/io/camunda/operate/webapp/api/v1/dao/opensearch/OpensearchSearchableDao.java#L63) and not in the query building logic.

🔧 Please validate this hypothesis first with @matthewBearCamunda.

🔧 As a general suggestion: the pull request currently contains 17 file changes in one commit, with not so much details in the message body about what has been exactly changed. Which makes it harder to review.
In the future, I'd recommend breaking those into smaller increments that tell a story, and add more details in the body.
For example:
- One commit that adds the new `collectQueryTerms method` with empty implementation
- One that implements it for the different DAOs
- One that starts using it in the buildFiltering method
- One that updates the tests
- etc

This is just an example, it doesn't have to be exactly like that

Christian-Broomfield committed:
test: added IT verifying deleted tenants do not appear in search results

mustafadagher committed:
test: update v1 tenancy test to use the existing `ProcessInstanceResult`

mustafadagher committed:
test: reuse helper methods from `TestHelper` util

mustafadagher committed:
test: wait for tenant deletion before asserting search results

mustafadagher committed:
refactor: get rid of the zeebe-util `VisibleForTesting` annotation

Also converted the method to be `package private`,
since that's enough visibility for tests

mustafadagher committed:
test: also wait for initial search results

To resolve current flakiness
